src/lib/helpers.ts

```diff
@@ -85,20 +84,13 @@ export const stripTags = (str?: string) => {
 export const markdownToText = str => {
   return stripTags(formatMarkdownValue(str, "html"))
 }
-export const parseFieldASTsIntoArray = fieldASTs => {
-  // TODO: when fieldASTs[*].selectionSet.selections.kind === "InlineFragment" recursively go deeper
-  return map(flatMap(fieldASTs, "selectionSet.selections"), "name.value")
-}
+
 export const queriedForFieldsOtherThanBlacklisted = (
-  fieldASTs,
+  info: GraphQLResolveInfo,
   blacklistedFields
 ) => {
-  if (!fieldASTs) return true
-  const queriedFields = parseFieldASTsIntoArray(fieldASTs)
-  return difference(queriedFields, blacklistedFields).length > 0
-}
-export const queryContainsField = (fieldASTs, soughtField) => {
-  return parseFieldASTsIntoArray(fieldASTs).includes(soughtField)
+  if (!info) return true
+  return includesFieldsOtherThanSelectionSet(info, blacklistedFields)
```

Comment:
Did not have time to go into too much details but stealing @joeyAghion usual comment here. I think we had a discussion how `blacklisted` is not a preferred terminology now. Maybe something with `deny` is a better option?